shell/platform/android/io/flutter/embedding/android/FlutterView.java

```diff
@@ -585,10 +600,42 @@ public void detachFromFlutterEngine() {
 //    }
   }
 
-  private boolean isAttachedToFlutterEngine() {
+  /**
+   * Returns true if this {@code FlutterView} is currently attached to a {@link FlutterEngine}.
+   */
+  @VisibleForTesting
```

Comment:
Why this annotation for these new members?

Reply:
I'm not aware of a use-case that requires this functionality outside of Espresso testing. To avoid unnecessary public API surface I added that annotation until such a time that a non-testing use-case appears.